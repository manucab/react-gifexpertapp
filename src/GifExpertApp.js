import React, {useState} from "react";
import AddCategory from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";
//rafc + tab

export const GifExpertApp = ( {defaultCategories = []} ) => {

    //const categories = ['One Punch', 'Samurai X', 'Dragon Ball']
    const [categories, setCategories] = useState( defaultCategories )

    // const handleAdd = () => {

    //     //setCategories([...categories,'Death Note'])
    //     setCategories( cats => [...cats, 'Death Note'])

    // }

    return (

        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={ setCategories }/>
            <hr/>

            <ol>
                {
                    categories.map( (c) => ( 
                        <GifGrid 
                            key={c}
                            category={ c } 
                        />
                    ))
                }
            </ol>

        </>

    )


}