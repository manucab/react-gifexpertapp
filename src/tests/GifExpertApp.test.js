import React from "react";

import { shallow } from "enzyme";
import { GifExpertApp } from "../GifExpertApp";

describe("Pruebas en GifExpertApp", () => {
  let wrapper;

  test("Debe mostrar la app correctamente", () => {
    wrapper = shallow(<GifExpertApp />);

    expect(wrapper).toMatchSnapshot();
  });

  test('debe de mostrar una lista de categorias', () => {
      
    const categories = ['One Punch', 'Full Metal Alchimist']
    wrapper = shallow(<GifExpertApp defaultCategories={ categories }/>);
    const el = wrapper.find('GifGrid').length

    expect(wrapper).toMatchSnapshot();
    expect( el ).toBe( categories.length )


  })
  


});
