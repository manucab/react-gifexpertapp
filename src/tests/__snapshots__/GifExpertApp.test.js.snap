// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp Debe mostrar la app correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en GifExpertApp debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Full Metal Alchimist"
      key="Full Metal Alchimist"
    />
  </ol>
</Fragment>
`;
